### ISSUE TITLE
Potential bug: lncli deletepayments --all, doesn't seem to work on v0.14.1-beta

### ISSUE THREAD

**Issue by miketwenty1** [lightningnetwork/lnd#6094]
### Background

Was recommended by @Roasbeef to get rid of failed payments as this could potentially help with cpu usage on LND.

### Your environment

* version of `lnd` 
v0.14.1-beta
* which operating system (`uname -a` on *Nix)
20.04 Ubuntu
* version of `btcd`, `bitcoind`, or other backend
bitcoind
* any other relevant environment details

### Steps to reproduce

Tell us how to reproduce this issue. Please provide stacktraces and links to code in question.
- run `lncli deletepayments --all`
- noticed I still get back failed payments with `lncli listpayments --include_incomplete ...` I still get the same amount of payments listed.
- no changes, so now restart node and force compaction
- check again, no changes with `lncli listpayments --include_incomplete ...`

### Expected behaviour

Failed payments such as some payments that have failure reason: "FAILURE_REASON_NO_ROUTE" should be gone.

### Actual behaviour

Nothing changed after running `lncli deletepayments --all`

**Comment by Roasbeef** [comment]
Just to be clear, we devised deleting the _failed_ payments, and not _all_ the payments. Hopefully you have the payments stored elsewhere if you need that data for accounting purposes.

**Comment by miketwenty1** [comment]
i saved off all the payments ahead of time just in case. However,.. running `lncli deletepayments --help` you see:

```
  If the --all flag is used, then all failed payments are removed. If so
  desired, _ALL_ payments (even the successful ones) can be deleted
  by additionally specifying --include_non_failed.
```

Am I not understanding something about `lncli deletepayments --all`?

**Comment by Roasbeef** [comment]
Ah no, my bad, looks like we modify the args/help to make it harder to shoot yourself in the foot!

**Comment by miketwenty1** [comment]
so, is this a bug?

**Comment by Roasbeef** [comment]
Are you properly paginating through the entire range? By default, you'll always get back the same numebr of payments there if you aren't paginating.

**Comment by miketwenty1** [comment]
i'm paginating 100,000 at a time with commands like this:

`lncli listpayments --include_incomplete --index_offset 200000 --max_payments 100000 --paginate_forwards true > z0_200000.json`

my question is.. after running `lncli deletepayments --all` shouldn't i not see anymore failed payments?

**Comment by hazrulnizam** [comment]
Yes, after running `lncli deletepayments --all` failed payments should not appear.

However, failed HTLCs of successful payments will still appear. Are you sure this is not what you are seeing?

**Comment by miketwenty1** [comment]
@hazrulnizam above for instance i still see, "FAILURE_REASON_NO_ROUTE" as a failed payment. 

Is this a bug?

**Comment by guggero** [comment]
You shouldn't see payments with `FAILURE_REASON_NO_ROUTE` anymore after running the `lncli deletepayments --all` command. I just tried this locally and it worked for me.

Do you see this in lnd's log?
`RPCS: [DeleteAllPayments] failed_payments_only=true, failed_htlcs_only=false`

Is it possible this just takes a very long time and is still ongoing? Or was, when you listed the payments again?
Or maybe it ran into a DB transaction timeout and rolled back the change?

If there are a _lot_ of payments, this can take a long time. Perhaps it helps if you first run `lncli deletepayments --all --failed_htlcs_only` and afterwards again without the second flag?

**Comment by miketwenty1** [comment]
i didn't need to run `lncli deletepayments --all --failed_htlcs_only` 
upon looking at this further.. the reason i saw a failure was because there was a fail that came immediately after i purged the failed payments. 

i tested this on another node just the "`lncli deletepayments --all`" and it works fine. this is not a bug, i'm going to close this, thanks!
